In [2]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [24]:
df = pd.read_csv('pima-indians-diabetes.csv',names=("pregnant","glucose concentration","blood pressure","thickness","serum insulin","BMI","Diabetes","Age","Class"))
df.head()

pregnant  glucose concentration  blood pressure  thickness  serum insulin  \
0         6                    148              72         35              0   
1         1                     85              66         29              0   
2         8                    183              64          0              0   
3         1                     89              66         23             94   
4         0                    137              40         35            168   

    BMI  Diabetes  Age  Class  
0  33.6     0.627   50      1  
1  26.6     0.351   31      0  
2  23.3     0.672   32      1  
3  28.1     0.167   21      0  
4  43.1     2.288   33      1

In [25]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
pregnant                 768 non-null int64
glucose concentration    768 non-null int64
blood pressure           768 non-null int64
thickness                768 non-null int64
serum insulin            768 non-null int64
BMI                      768 non-null float64
Diabetes                 768 non-null float64
Age                      768 non-null int64
Class                    768 non-null int64
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [28]:
df.shape

(768, 9)

1. Title: Pima Indians Diabetes Database
2. Sources:
   (a) Original owners: National Institute of Diabetes and Digestive and
                        Kidney Diseases
   (b) Donor of database: Vincent Sigillito (vgs@aplcen.apl.jhu.edu)
                          Research Center, RMI Group Leader
                          Applied Physics Laboratory
                          The Johns Hopkins University
                          Johns Hopkins Road
                          Laurel, MD 20707
                          (301) 953-6231
   (c) Date received: 9 May 1990
3. Past Usage:
    1. Smith,~J.~W., Everhart,~J.~E., Dickson,~W.~C., Knowler,~W.~C., \&
       Johannes,~R.~S. (1988). Using the ADAP learning algorithm to forecast
       the onset of diabetes mellitus.  In {\it Proceedings of the Symposium
       on Computer Applications and Medical Care} (pp. 261--265).  IEEE
       Computer Society Press.
       The diagnostic, binary-valued variable investigated is whether the
       patient shows signs of diabetes according to World Health Organization
       criteria (i.e., if the 2 hour post-load plasma glucose was at least 
       200 mg/dl at any survey  examination or if found during routine medical
       care).   The population lives near Phoenix, Arizona, USA.
       Results: Their ADAP algorithm makes a real-valued prediction between
       0 and 1.  This was transformed into a binary decision using a cutoff of 
       0.448.  Using 576 training instances, the sensitivity and specificity
       of their algorithm was 76% on the remaining 192 instances.
4. Relevant Information:
      Several constraints were placed on the selection of these instances from
      a larger database.  In particular, all patients here are females at
      least 21 years old of Pima Indian heritage.  ADAP is an adaptive learning
      routine that generates and executes digital analogs of perceptron-like
      devices.  It is a unique algorithm; see the paper for details.
5. Number of Instances: 768
6. Number of Attributes: 8 plus class 
7. For Each Attribute: (all numeric-valued)
   1. Number of times pregnant
   2. Plasma glucose concentration a 2 hours in an oral glucose tolerance test
   3. Diastolic blood pressure (mm Hg)
   4. Triceps skin fold thickness (mm)
   5. 2-Hour serum insulin (mu U/ml)
   6. Body mass index (weight in kg/(height in m)^2)
   7. Diabetes pedigree function
   8. Age (years)
   9. Class variable (0 or 1)
8. Missing Attribute Values: Yes
9. Class Distribution: (class value 1 is interpreted as "tested positive for
   diabetes")
   Class Value  Number of instances
   0            500
   1            268
10. Brief statistical analysis:

    Attribute number:   
       Mean:   Standard Deviation:
    1.                     3.8     3.4
    2.                   120.9    32.0
    3.                    69.1    19.4
    4.                    20.5    16.0
    5.                    79.8   115.2
    6.                    32.0     7.9
    7.                     0.5     0.3
    8.                    33.2    11.8

In [33]:
# select features from the dataset  for model building purpose
feature_select_col=['pregnant','serum insulin', 'BMI', 'Age']

In [35]:
# create features object
X_features=df[feature_select_col]

In [39]:
# create response object
Y_target=df['Class']

In [42]:
# split test and training data
# by default 75% training data and 25% testing data
from sklearn.model_selection import train_test_split
# code to split the govven dataset into training and testing datasets
x_train,x_test, y_train,y_test=train_test_split(X_features, Y_target, random_state=1)

In [46]:
# import logisitc regression model on the traing set
from sklearn.linear_model import LogisticRegression
logReg=LogisticRegression()
logReg.fit(x_train,y_train)

LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
          intercept_scaling=1, max_iter=100, multi_class='ovr', n_jobs=1,
          penalty='l2', random_state=None, solver='liblinear', tol=0.0001,
          verbose=0, warm_start=False)

In [48]:
# make predictions for the testing set
y_pred=logReg.predict(x_test)
# check accuecy
# use sklearn metrics for accuracy
from sklearn import metrics
metrics.accuracy_score(y_test,y_pred)

0.6979166666666666

In [53]:
# print the first 30 true and predicted responses
print('actual', y_test.values[0:30])
print('predicted', y_pred[0:30])

actual [0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 1 1 0 0 0 1 1 1 1 0 0 0 1 0 1]
predicted [0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0]
